src/tests/krate.rs

```diff
@@ -67,20 +67,28 @@ struct SummaryResponse {
     popular_categories: Vec<EncodableCategory>,
 }
 
+impl ::util::MockTokenUser {
```

Comment:
Eeeeeenteresting. I'm not sure how I feel about these test-file-local `impl`s. My worry is that there isn't one place to go to see all the actions that are nicely implemented for you to call in tests (and see what the method names are and how to call them), which could potentially lead to multiple implementations of the same functionality.... however, if there are actions that are really only relevant to one file of tests, then that gives you less you need to look through, if you look at the globally available actions and the locally available actions.

Typing this out, I'm wondering if maybe this will push us to better test organization-- create a file that's only about yanking crates, say, and this function will only be needed there, because not all the tests in this file have to do with yanking.

Hhhhmmmm. I'm mostly just rambling, let's keep this and see how it works out over time.

Reply:
For the 2 files where I did this, I did a grep on the format strings and I'm pretty sure these won't be needed elsewhere, at least for our current suite of tests.  I agree we want to be careful not to duplicate these in multiple files.

The other downside is that the impl is on only one of the 3 concrete types.  So there is a lot of motivation to move this to the trait if the functionality is generally useful.